src/java.desktop/macosx/classes/sun/awt/CGraphicsDevice.java

```diff
@@ -65,9 +65,11 @@
 
     // Save/restore DisplayMode for the Full Screen mode
     private DisplayMode originalMode;
+    private DisplayMode initialMode;
 
     public CGraphicsDevice(final int displayID) {
         this.displayID = displayID;
+        this.initialMode = getDisplayMode();
```

Comment:
This probably should be revalidated when this device is invalidated, otherwise deleted device will restore its own old-initial mode, instead of the new-mode for the new device.

Reply:
Ok I've added copying it in invalidate()